### ISSUE TITLE
toFixed(0) is broken on some browsers (fix proposition)

### ISSUE THREAD

**Issue by karczk** [BenjaminVanRyseghem/numbro#317]
**Test case:**
`numbro(123456789123457000).format(0,0)`

**Actual result (invalid):**
"123,456,789,123,456,992"
Broken on Chrome and Firefox, but works on IE/Edge.
(en-US but this is not important)

**Expected result:**
"123,456,789,123,457,000"

**Source of problem (Numbro source):**
`output = (roundingFunction(value + 'e+' + precision) / power).toFixed(precision);`

toFixed(0) has broken accurancy:
Console: `(123456789123457000).toFixed(0)`

Chrome, Firefox: "123456789123456992" (broken)
IE11/Edge: "123456789123457000"

**Solution:**
```
output = roundingFunction(value + 'e+' + precision) / power;
output = precision === 0 ? (Math.round(output) + '') : output.toFixed(precision);
```
Math.round is more accurate and compatible e.g. with C#. toFixed(0) is not cross browser.

**Other example**
Input: 123456789123457000.34535

Compatible on different platforms
C# (Math.Round) -> 123456789123457010
IE, Chrome, Firefox (Math.round) -> 123456789123457010

IE (toFixed(0)) -> "123456789123457000"
FF, Chrome (toFixed(0)) -> "123456789123457008"

**Comment by BenjaminVanRyseghem** [comment]
thanks :smile:

have you tried it in the `2.0.0` branch by chance? This new branch uses `BigNumber.js` to handle those situations, with hopefully a better output

**Comment by karczk** [comment]
No, I didn't, Good to know that new big version is coming :) If you could add the above fix in next release, I would be grateful.

**Comment by karczk** [comment]
@BenjaminVanRyseghem, it works on 2.0.0, test cases:
```
[123456789123457000, "0,0", "123,456,789,123,457,000"],
[123456789123457000.34535, "0,0", "123,456,789,123,457,010"]
```
Still, it should be fixed on 1.x :)

**Comment by BenjaminVanRyseghem** [comment]
@karczk unfortunately, it's very time consuming to fix code in 1.x, and I prefer to focus on 2.0 😕
